lib/snyk/enrich_spdx.go

```diff
@@ -35,6 +36,47 @@ const (
 	snykVulnerabilityDB_URI = "https://security.snyk.io"
 )
 
+type spdxEnricher = func(*spdx_2_3.Package, *packageurl.PackageURL)
+
+var spdxEnrichers = []spdxEnricher{
+	enrichSPDXSnykAdvisorData,
+	enrichSPDXSnykVulnerabilityDBData,
+}
+
+func enrichSPDXSnykAdvisorData(component *spdx_2_3.Package, purl *packageurl.PackageURL) {
+	url := SnykAdvisorURL(purl)
+	if url != "" {
+		ext := &spdx_2_3.PackageExternalReference{
+			Locator:            url,
+			RefType:            "advisory",
+			Category:           "Other",
+			ExternalRefComment: "Snyk Advisor",
+		}
+		if component.PackageExternalReferences == nil {
+			component.PackageExternalReferences = []*spdx_2_3.PackageExternalReference{ext}
+		} else {
+			component.PackageExternalReferences = append(component.PackageExternalReferences, ext)
+		}
+	}
+}
+
+func enrichSPDXSnykVulnerabilityDBData(component *spdx_2_3.Package, purl *packageurl.PackageURL) {
+	url := SnykVulnURL(purl)
+	if url != "" {
+		ext := &spdx_2_3.PackageExternalReference{
+			Locator:            url,
+			RefType:            "url",
+			Category:           "Other",
```

Comment:
i had totally forgotten [about this PR](https://github.com/spdx/tools-golang/pull/229), but looks like we could potentially use a constant for this if we upgrade spdx tools to latest.